src/authentication_manager.rs

```diff
@@ -22,7 +22,8 @@ impl AuthenticationManager {
     pub async fn get_token(&self, scopes: &[&str]) -> Result<Token, GCPAuthError> {
         let mut sa = self.service_account.lock().await;
         let mut token = sa.get_token(scopes);
-        if token.is_none() {
+
+        if token.is_none() || token.clone().unwrap().has_expired() {
```

Comment:
The current code is just refactor as it keeps the same functionality (returning library error after refresh returned invalid token)